src/analysis/analysis-poller.js

```diff
@@ -1,22 +1,32 @@
 var _ = require('underscore');
 var BackbonePoller = require('backbone-poller');
 
-var AnalysisPoller = function () {};
+var AnalysisPoller = function () {
+  this._pollers = [];
+};
 
 AnalysisPoller.CONFIG = {
   START_DELAY: 1000,
   MAX_DELAY: Infinity,
   DELAY_MULTIPLIER: 1.5
 };
 
-AnalysisPoller.prototype.poll = function (analysisModel) {
-  this._pollers = [];
-  var poller = this._findOrCreatePoller(analysisModel);
-  if (analysisModel.hasChanged('url') && poller.active()) {
-    poller.stop();
-  }
-  if (!analysisModel.isDone()) {
-    poller.start();
+AnalysisPoller.prototype.resetAnalysisNodes = function (analysisModels) {
+  this.reset();
+  _.each(analysisModels, function (analysisModel) {
+    this._poll(analysisModel);
+  }, this);
+};
+
+AnalysisPoller.prototype._poll = function (analysisModel) {
+  if (analysisModel.url()) {
+    var poller = this._findOrCreatePoller(analysisModel);
+    if (analysisModel.hasChanged('url') && poller.active()) {
+      poller.stop();
```

Comment:
why are we stopping the poller in this case?

Reply:
This means there is an active poller for this analysisModel to a previous url. Since we now have a new url, we have to stop the previous poller and create a new one.

Reply:
Since is a little tricky maybe we could add this exact answer as a comment in the code